src/Integration.Vsix/SonarLintDaemon/CFamily.cs

```diff
@@ -34,15 +34,15 @@ static class CFamily
         internal const string CPP_LANGUAGE_KEY = "cpp";
         internal const string C_LANGUAGE_KEY = "c";
 
-        public static string TryGetConfig(EnvDTE.ProjectItem projectItem, string absoluteFilePath, out string sqLanguage)
+        public static string TryGetConfig(ISonarLintOutput logger, EnvDTE.ProjectItem projectItem, string absoluteFilePath, out string sqLanguage)
         {
             try
             {
                 return FileConfig.TryGet(projectItem, absoluteFilePath).ToJson(absoluteFilePath, out sqLanguage);
             }
             catch (Exception e)
             {
-                VsShellUtils.WriteToSonarLintOutputPane(ServiceProvider.GlobalProvider, "Unable to collect C/C++ configuration: " + e.ToString());
+                logger.Write("Unable to collect C/C++ configuration: " + e.ToString());
```

Comment:
Miscellaneous: _ISonarLintOutput_ was introduced as an abstraction so classes don't need to write to the output window directly. I've changed calls from _VsShellUtils.WriteToSonarLintOutputPane_ to _logger.Write_ in a couple of places.

Reply:
+1 (we should try to do this everywhere. Static classes are a nightmare to mock)